### PR TITLE
[17.05] Fix setting up explicit container resolution...

### PR DESCRIPTION
**Pull request by jmchilton**
... from a container resolution config file.

### CODE REVIEW

File: lib/galaxy/util/plugin_config.py [hunk 1]
```diff
@@ -45,7 +45,14 @@ def __load_plugins_from_element(plugins_dict, plugins_element, extra_kwds):
...
         plugin_kwds.update( extra_kwds )
-        plugin = plugins_dict[ plugin_type ]( **plugin_kwds )
+        try:
+            plugin_klazz = plugins_dict[ plugin_type ]
+        except KeyError:
+            template = "Failed to find plugin of type [%s] in available plugin types %s"
+            message = template % ( plugin_type, str( plugins_dict.keys() ) )
+            raise Exception( message )
+
+        plugin = plugin_klazz( **plugin_kwds )
```
Comment (bgruening) [inline lib/galaxy/util/plugin_config.py:55]: `plugin_klazz` is not defined if the KeyError exception is raised, isn't it?

Comment (jmchilton) [reply lib/galaxy/util/plugin_config.py:55]: If the ``KeyError`` is raised - I raise a new more informative exception so the code would never reach this point.

Comment (bgruening) [reply lib/galaxy/util/plugin_config.py:55]: True!

**Comment by bgruening** [comment]
:+1:

**Pull request merged**